openpmd_viewer/openpmd_timeseries/data_reader/io_reader/utilities.py

```diff
@@ -56,17 +57,19 @@ def get_data(series, record_component, i_slice=None, pos_slice=None,
     if i_slice is not None and not isinstance(i_slice, list):
         i_slice = [i_slice]
 
-    chunks = record_component.available_chunks()
-
+    # read whole data set
     if pos_slice is None:
+        print(f"record_component: {record_component}, {record_component.shape}, {record_component.dtype}")
         # mask invalid regions with NaN
         data = np.full_like(record_component, np.nan)
         for chunk in chunks:
             chunk_slice = chunk_to_slice(chunk)
+            print(f"chunk: {chunk}, {chunk_slice}")
             # read only valid region
             x = record_component[chunk_slice]
```

Comment:
I wonder if this is related to https://github.com/openPMD/openPMD-api/pull/1225
Prior to #332 we did not copy an extra time.

Reply:
actually, @AlexanderSinn found that `x = record_component[()]` is fast, but it becomes slow the moment we have the `data = np.full_like(record_component, np.nan)` above.

We could try to make the above line a `np.full` with shape and dtype.

Reply:
actually, @AlexanderSinn found that `x = record_component[()]` is fast, but it becomes slow the moment we have the `data = np.full_like(record_component, np.nan)` above.

We could try to make the above line a `np.full` with shape and dtype.

Reply:
@AlexanderSinn found `data = np.full(record_component.shape, np.nan, record_component.dtype)` solves this! :star_struck: 

That's probably why meshes below (`data = np.full`) were not affected.